token/services/ttx/convert.go

```diff
@@ -0,0 +1,271 @@
+	}
+	result := resultBoxed.([]interface{})
+	ir := result[0].(*ConversionRequest)
+	return ir.RecipientData.Identity, result[1].(view.Session), nil
+}
+
+func (r *ConversionInitiatorView) Call(context view.Context) (interface{}, error) {
+	span := context.StartSpan("conversion_request_view")
+	defer span.End()
+	logger.Debugf("Respond request recipient identity using wallet [%s]", r.Wallet)
+
+	span.AddEvent("start_session")
+	session, err := session.NewJSON(context, context.Initiator(), r.Issuer)
+	if err != nil {
+		logger.Errorf("failed to get session to [%s]: [%s]", r.Issuer, err)
+		return nil, errors.Wrapf(err, "failed to get session to [%s]", r.Issuer)
+	}
+
+	// first agreement
+	agreement := &ConversionAgreement{}
+	span.AddEvent("send_conversion_agreement")
+	err = session.SendWithContext(context.Context(), agreement)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to send recipient data")
+	}
+
+	if err := session.ReceiveWithTimeout(agreement, 1*time.Minute); err != nil {
+		return nil, errors.Wrapf(err, "failed to receive conversion agreement")
+	}
+
+	// prepare request
+	span.AddEvent("send_conversion_request")
+
+	// - recipient identity
+	tmsID, recipientIdentity, auditInfo, tokenMetadata, err := r.getRecipientIdentity(context)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get recipient identity")
+	}
+	// - proof
+	tms := token.GetManagementService(context, token.WithTMSID(*tmsID))
+	if tms == nil {
+		return nil, errors.Errorf("tms not found for [%s]", tmsID)
+	}
+	proof, err := tms.TokensService().GenConversionProof(agreement.Challenge, r.Tokens)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to generate proof")
+	}
+
+	wr := &ConversionRequest{
+		ID:    agreement.Challenge,
+		TMSID: *tmsID,
+		RecipientData: RecipientData{
+			Identity:      recipientIdentity,
+			AuditInfo:     auditInfo,
+			TokenMetadata: tokenMetadata,
+		},
+		Tokens:       r.Tokens,
+		Proof:        proof,
+		NotAnonymous: r.NotAnonymous,
+	}
+	err = session.SendWithContext(context.Context(), wr)
+	if err != nil {
+		logger.Errorf("failed to send recipient data: [%s]", err)
+		return nil, errors.Wrapf(err, "failed to send recipient data")
+	}
+
+	return []interface{}{wr, session.Session()}, nil
+}
+
+// WithWallet sets the wallet to use to retrieve a recipient identity if it has not been passed already
+func (r *ConversionInitiatorView) WithWallet(wallet string) *ConversionInitiatorView {
+	r.Wallet = wallet
+	return r
+}
+
+// WithTMSID sets the TMS ID to be used
+func (r *ConversionInitiatorView) WithTMSID(id token.TMSID) *ConversionInitiatorView {
+	r.TMSID = id
+	return r
+}
+
+// WithRecipientData sets the recipient data to use
+func (r *ConversionInitiatorView) WithRecipientData(data *RecipientData) *ConversionInitiatorView {
+	r.RecipientData = data
+	return r
+}
+
+func (r *ConversionInitiatorView) getRecipientIdentity(context view.Context) (*token.TMSID, view.Identity, []byte, []byte, error) {
```

Comment:
Maybe rename this method. Or otherwise inline it. Its logic is pretty straightforward and it has too many outputs.

Reply:
but inlining makes the calling method bigger, no?

Reply:
It's a matter of taste. They are simple calls in this method without any data calculations. If they had more calculations that all produce one result, I would definitely extract them. But still whatever is better for you. In this case we can just rename this method to reflect what it gets.